### ISSUE TITLE
Completely broken in thunderbird

### ISSUE THREAD

**Issue by jrpear** [glacambre/firenvim#984]
OS Version: linux x86-64
Browser Version: Tried Thunderbird 88.0a1 and 87.0b2
Browser Addon Version: 0.2.4
Neovim Plugin Version: 0.2.4

## What I tried to do:

* Install the firenvim plugin to thunderbird and neovim
* Open the compose email window
* Do neovim stuff

## What happened:

* Cursor started at bottom of body field, where I would expect vim status line to be
* Began typing
* Characters appeared at top of body field, where they would normally go without firenvim, cursor also moves up there
* Press esc to exit insert mode
* Nothing happens. Nothing vim-related seems to work.
* Try to click and drag with mouse to highlight some text
* Mouse begins dragging blank white image the size of the body field

Also, when I swap composition mode from plaintext to HTML, I see that the "blank white image" is actually an iframe.

Basically, it seems like it's being loaded but something is very wrong. I've gone through the steps outlined in TROUBLESHOOTING, and reduced my init.vim to 3 lines, to no avail:

```
call plug#begin()
Plug 'glacambre/firenvim', { 'do': { _ -> firenvim#install(0) } }
call plug#end()
```

Here's a screenshot of the weird state it's in (I resized the window so the weird white box is very visible, it doesn't resize with the window):
![image](https://user-images.githubusercontent.com/26498132/110868812-1c774200-827e-11eb-999a-60db95a72478.png)

**Comment by glacambre** [comment]
Hi, thanks for the report. Thunderbird support is very experimental and I haven't had time to work on it in a while. Firenvim used to be usable with caveats there but things probably got worse with Thunderbird updates.

I'll take the add-on down from the thunderbird extension store but keep your issue up to remind myself to work on it :).

**Comment by jpear1** [comment]
Ah alright. Thanks for all your work on this great project!

**Comment by BlueDrink9** [comment]
Ah man, I clicked on this issue because I've recently started using Thunderbird, and I am missing my webUI firenvim something fierce. A shame to see support is difficult

**Comment by glacambre** [comment]
I think it's just a matter of writing a few thunderbird patches to fix edge cases that the thunderbird authors haven't thought about, there is no major api/architecture design to do.

So if you're looking for a weekend project, this might just be it! ;)

**Comment by BlueDrink9** [comment]
As much as I'd love to contribute to Firenvim, I know zero web stack techs, so it'd be a learning curve

**Comment by glacambre** [comment]
If you're afraid learning a web stack will turn you in a web developer and force you to work on websites until you retire, don't be afraid, this hasn't happened to me. Companies are still pretty happy to hire me as a compiler engineer even though I have Typescript on my resume :smile:.

**Comment by BlueDrink9** [comment]
Never afraid of learning, just concerned about the time it would take! "in and out, quick patch, we'll be back before you know it"

**Comment by glacambre** [comment]
This is a thunderbird bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1707449

I'm not very interested in fixing it myself as [last time I did that](https://bugzilla.mozilla.org/show_bug.cgi?id=1675589) I had a hard time convincing the thunderbird maintainers that I was doing something sane, so I'll let them do the work this time :).

**Comment by BlueDrink9** [comment]
Man, that's a real shame that they're souring the enthusiasm of such an excellent OSS community member

**Comment by glacambre** [comment]
I merged #1137, which finalizes support for Firenvim in thunderbird (hopefully this time things won't break out of the blue!). This currently requires using Thunderbird nightly (stable releases break Firenvim when replying to emails).

I'll publish Firenvim to thunderbird add-on store when I publish a new release for browsers :)

**Comment by jrpear** [comment]
Woot woot!

**Comment by BlueDrink9** [comment]
So stoked!

**Comment by glacambre** [comment]
Firenvim for thunderbird has been released: https://addons.thunderbird.net/en-US/thunderbird/addon/firenvim/

It's still pretty rough due to issues with Thunderbird's extension APIs, but I find it usable.

**Comment by jrpear** [comment]
Yay! Time to go try it out.
